Improve support msvc

reds-heig opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/giotto-ai/giotto-tda/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

#338 

#### What does this implement/fix? Explain your changes.

This PR fixes incompatible compilation flags when compiling wit MSVC.
Before this PR, compilers flags were not per compiler, this produced on MSVC warnings due to unknown compiler options.

Also, I performed some clean-ups to improve maintainability of `CMakeLists.txt`.

#### Any other comments?


<!--
We value all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.

Thanks for contributing!
-->

Commit:
Add support to MSVC flags at compilation

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

Commit:
Clean up CMakeLists.txt file

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

ulupo reviewed (commented):
Looks good to my ignorant eyes. Some code looks repeated but if it's not worth defining a single function to avoid this repetition, it's OK with me.

Commit:
Update ccache key to clear old cache

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

Commit:
Add /AVX compiler flag for MSVC

For more details on the flag, see https://docs.microsoft.com/en-us/cpp/build/reference/arch-x86\?view\=vs-2019

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

Commit:
Update ccache for CI

Signed-off-by: julian <julian.burellaperez@heig-vd.ch>

ulupo reviewed (approved):
Great!
